test/files/run/patmat-seq.check

```diff
@@ -66,7 +66,7 @@ package <empty> {
       case <synthetic> val x1: Int(2) = 2;
       case16(){
         <synthetic> val o18: collection.SeqFactory.UnapplySeqWrapper[Int] = A.unapplySeq(x1);
-        if (o18.isEmpty.unary_!)
+        if (false.unary_!)
```

Comment:
Btw, I fully expect this to be a pre-existing bug, basically unrelated to your PR here...

Reply:
I'd have to check.

Reply:
see the new tests: This doesn't seem to break anything, but I wouldn't know why.

Reply:
Hmm.  Maybe copy how `test/files/run/patmat-seq.scala` is setup to look at the trees.

Reply:
I'll give that a try, I'll ping you when I have something like that.

Reply:
I've added a new test along these lines, but I can't manage to actually get it to emit the same tree. Maybe it's special-cased for UnnaplySeqWrapper somewhere? If it is, I can't find where.

Reply:
The new test "shows" the call doesn't get elided. Unfortunately, I can't confirm that it only inlines the check if it's safe, since I can't get it to inline at all.

Reply:
Are you working on this?  Should I jump in?  I'm not 100% sure when 3.0.0-M4 or 3.0.0-RC1 are coming out, but we want to release 2.13.5 near it and I'm keen for these improvements to make it in.

Reply:
Can you take a look whether this is enough for you to satisfy the fear of a mis-inlined `isEmpty` side-effect? If not, do you have any idea how I can trigger that inlining seen in this testcase?

Reply:
Well that was painful...  and I'm not sure if the way I "fixed" it is reasonable/safe or not.